src/lib/components/SelectCharacter.svelte

```diff
@@ -1,15 +1,173 @@
 <script lang="ts">
+	import type { CharacterShort } from '$lib/model';
+	import { mutateCharacterShort } from '$lib/model';
+	import { onMount, onDestroy } from 'svelte';
+
 	export let redirectUrl: string = '/';
 	export let character: string = '';
+
+	let searchResults: CharacterShort[] = [];
+	let searching = false;
+	let searchIsFocused = false;
+	let debounceTimer: number;
+	let controller = new AbortController();
+
+	//Click handlers for autocomplete visibility
+	onMount(() => {
+		if (typeof window !== 'undefined') {
+			window.addEventListener('click', handleClickOutside);
+		}
+	});
+
+	//Destroy listener with component to avoid memleaks
+	onDestroy(() => {
+		if (typeof window !== 'undefined') {
+			window.removeEventListener('click', handleClickOutside);
+		}
+	});
+
+	function handleClickOutside() {
+		var searchField = document.getElementById('character');
+		searchIsFocused = document.activeElement === searchField;
+		if (searchIsFocused && searchResults?.length === 0) performSearch();
+	}
+
+	function debounce(func: () => void, delay: number) {
+		clearTimeout(debounceTimer);
+		debounceTimer = setTimeout(func, delay);
+	}
+
+	function splitByQuery(name: string) {
+		const regex = new RegExp(`(${character})`, 'gi');
+		//Find query string in the character name, split to parts and remove split "" artifacts
+		const parts = name
+			.toLowerCase()
+			.split(regex)
+			.filter((el) => el !== '');
+
+		//Return name with highlights as html
+		const highlighted = parts
+			.map((part, index) =>
+				part.toLowerCase() === character.toLowerCase()
+					? `<span style="color: var(--color-primary)">${
```

Comment:
Try to make a svelte component for this. You can always map this to some kind of a structure and pass it to the component.